src/lib/rustContext.ts

```diff
@@ -49,11 +50,25 @@ export default class RustContext {
   private projectId = 0
   public readonly planesCreated = new Signal()
 
+  private _settingsActor: SettingsActorType
+  get settingsActor() {
+    return this._settingsActor
+  }
+  set settingsActor(settingsActor: SettingsActorType) {
+    this._settingsActor = settingsActor
+  }
+
   constructor(
     engineCommandManager: ConnectionManager,
-    instance: Promise<ModuleType | string>
+    instance: Promise<ModuleType | string>,
+    /**
+     * TODO: move settings system upstream of KclManager so this hack isn't necessary.
```

Comment:
In the future it would look something like

```index.ts
// pesudocode
const wasmInstance = loadWasm();
const settingsFromDisk = fs.readFromDiskSyncSettings();
const jsSettings = wasmInstance.loadSettings(settingsFromDisk); // serialize/deserialize logic with serde
createSingletons();
createReact();
````

I am probably messing up the internals but we would need to load wasm, read `.toml` off disk and pass that over the boundary to parse it properly and get a JS object back before doing anything else with singletons?